contracts/presale/Presale.sol

```diff
@@ -70,4 +70,11 @@ contract Presale is Ownable, PresaleConstants, ReentrancyGuard {
     function purchasedAmount(address _account) external view returns (uint) {
         return purchasedAmountOf[_account];
     }
+
+    function presaleIsActive() external view returns (bool) {
+        if ((startBlock != 0) && (block.number <= startBlock + duration)) {
```

Comment:
```Solidity
function presaleIsActive() external view returns (bool) {
    return block.number >= startBlock && block.number <= startBlock + duration;
}
```

Reply:
I still want to add the check startBlock != 0. By default, before start Presale startBlock = 0, and if duration is a very large number (equivalent to 6+ years), then method presaleIsActive return true even when Presale has not started yet.

In this case, the presaleIsActive method is shown as follows:
```
function presaleIsActive() external view returns (bool) {
        return startBlock != 0 && block.number >= startBlock && block.number <= startBlock + duration;
}
```